### ISSUE TITLE
Uniform unit tests

### ISSUE THREAD

**Issue by shirk3y** [scrapinghub/dateparser#31]
It seems that we have plenty of methodologies used in unit tests.
Some of them are:

1.

        date = DateParser(language='cz').parse('pon 16. čer 2014 10:07:43')
        self.assertEqual(date.year, 2014)
        self.assertEqual(date.month, 6)
        self.assertEqual(date.day, 16)
        self.assertEqual(date.hour, 10)
        self.assertEqual(date.minute, 07)
        self.assertEqual(date.second, 43)

2.

       parser = DateParser()
        date_fixtures = [
            ('13 iunie 2013', datetime(2013, 6, 13)),
            ('14 aprilie 2014', datetime(2014, 4, 14)),
            ('18 martie 2012', datetime(2012, 3, 18)),
        ]

        for dt_string, correct_date in date_fixtures:
            parsed = parser.parse(dt_string)
            self.assertEquals(correct_date.date(), parsed.date())

3.

        @parameterized.expand([
        param('Sep 03 2014 | 4:32 pm EDT', datetime(2014, 9, 3, 21, 32)),
        param('17th October, 2034 @ 01:08 am PDT', datetime(2034, 10, 17, 9, 8)),
        param('15 May 2004 23:24 EDT', datetime(2004, 5, 16, 4, 24)),
        param('15 May 2004', datetime(2004, 5, 15, 0, 0)),
        param('Nov 25 2014 10:17 pm EST', datetime(2014, 11, 26, 4, 17)),
        ])

4.

       date = DateParser(language='pl').parse('Środa, 26 listopada 2014 10:11:12')
       self.assertEqual(date.timetuple()[:6], (2014, 11, 26, 10, 11, 12))

Maybe we could create simple method/function for asserting correct date? Eg.
        
        date = DateParser(language='en').parse('Tue, 25 Dec, 2012 12:00')
        self.assertDate(date, 2012, 12, 25, 12, 0)

Depending of number of params given to `assertDate` we check date with given resolution

**Comment by horva** [comment]
What about having python file having key: value entries, key representing arg of dateparser.parse and value the result. Then we can iterate through all the entries within single test, asserting with dateparser.parse, which would make adding new cases much easier.

Suggested format:

    from datetime import datetime
    TODAY = datetime(2015, 1, 1)  # for freshness and other relative calculations

    TEST_DATA = {
        'en': {
            'sep 19': datetime(2014, 9, 19),
            ...
        },
        'other': {
            '2015 02 03': datetime(2015, 2, 3),
            ...
        },
    }

**Comment by Allactaga** [comment]
Iterating data in single test is bad idea of massive proportions. Please use `@parameterized.expand`. It combines clarity and repeatability.

**Comment by Allactaga** [comment]
All existing tests should be reviewed and changed to the given-when-then approach. Here is a good [example](https://github.com/scrapinghub/dateparser/pull/26/files#diff-36128d81467a740315c9e5f49860c291R182) how it should look in the end:
```python

    def test_dates_parsing(self, date_string, expected):
        self.given_utcnow(datetime(2012, 11, 13))  # Tuesday
        self.given_local_tz_offset(0)
        self.given_parser()
        self.given_date_string(date_string)
        self.when_date_is_parsed()
        self.then_date_was_parsed_by_date_parser()
        self.then_period_is('day')
        self.then_date_obj_exactly_is(expected)
```
All cases should be inherited from `BaseTestCase` and use `@parameterized.expand` for the sets of data.

**Comment by Allactaga** [comment]
For those who would take this issue:
Some tests is a little bit out of the 'easy' scope of this ticket and require deeper understanding of the code in this project. I am talking about tests described in [this issue](https://github.com/scrapinghub/dateparser/issues/44). But if you fill up to it, please feel free to work on those tests too (just don't forget to assign both issues to yourself).

**Comment by Allactaga** [comment]
Commit https://github.com/scrapinghub/dateparser/commit/11e322cf99e0cd38cce1874c564cc59ad04885ad could be used as an example of creating inherited test cases

**Comment by tsrdatatech** [comment]
I am going to take this issue.

**Comment by waqasshabbir** [comment]
Fixed by #85
